apps/aens/src/aens_utils.erl

```diff
@@ -86,22 +72,18 @@ validate_name(Name) ->
             {error, multiple_namespaces}
     end.
 
-validate_name_ascii(NameAscii) ->
-    case length(NameAscii) > 253 of
-        true  -> {error, name_too_long};
-        false ->
-            Labels = string:split(NameAscii, ".", all),
+name_to_ascii(Name) when is_binary(Name) ->
+    NameUnicodeList = unicode:characters_to_list(Name, utf8),
+    try idna:encode(NameUnicodeList, [{uts46, true}, {std3_rules, true}]) of
+        NameAscii ->
             %% idna:to_ascii(".aet") returns just "aet"
-            case length(Labels) of
-                1  -> {error, no_label_in_registrar};
-                _N -> validate_labels(Labels)
+            case length(string:split(NameAscii, ".", all)) =:= 1 of
+                true  -> {error, no_label_in_registrar};
+                false -> {ok, list_to_binary(NameAscii)}
             end
-    end.
-
-validate_labels([]) ->
-    ok;
-validate_labels([Label | Rest]) ->
-    case length(Label) > 0 andalso length(Label) =< 63 of
-        true  -> validate_labels(Rest);
-        false -> {error, bad_label_length}
+    catch
+        exit:{bad_label, Reason} ->
+            {error, {bad_label, Reason}};
+        exit:{invalid_codepoint, Cp} ->
+            {error, {invalid_codepoint, Cp}}
```

Comment:
please add tests to cover those errors outside the lib

Reply:
done

Reply:
@sennui Please revisit.